File: Recursion/JavaScript/checkPalindrome.js
let str = 'racecar'

function checkPalindrome(arr,i,j){
    if(i>j){
        return true
    }

    if(arr[i] !== arr[j]){
        return false
    }

    return checkPalindrome(arr,i+1,j-1);
}

console.log(checkPalindrome(str,0,str.length-1));

File: Recursion/Python/checkPalindrome.py
str = list('racecar')

def checkPalindrome(arr,i,j):
    if i>j:
        return True
    
    if arr[i] != arr[j]:
        return False

    return checkPalindrome(arr, i+1, j-1)


print(checkPalindrome(str, 0, len(str)-1))